### ISSUE TITLE
Video records time not match

### ISSUE THREAD

**Issue by maheshgiri** [twilio/twilio-video.js#920]
I am using Twilio  video js 2.0.1 version
This is the code that I am using
`
roomJoined = room => {
    const _self = this;
    setTimeout(() => {
      _self.setState({ loadingbutton: false });
      console.log('state changedd ');
    }, 15000);

    // _self.setState({ loadingbutton: false });
    console.log('State changed ' + this.state.loadingbutton + _self.state.loadingbutton);
    this.activeRoom = room;
    window.room = room.name;

    //this.log("Joined as '" + this.identity + "'");
    //this.refs.buttonJoin.style.display = "none";
    //this.refs.buttonLeave.style.display = "inline";

    // Attach LocalParticipant's Tracks, if not already attached.
    var previewContainer = this.refs.localMedia;
    var previewContainer = document.getElementById('local-media');
    if (!previewContainer.querySelector('video')) {
      this.attachParticipantTracks(room.localParticipant, previewContainer);
    }

    // Attach the Tracks of the Room's Participants.
    room.participants.forEach(participant => {
      console.log("Already in Room: '" + participant.identity + "'");
      var previewContainer = document.getElementById('remote-media');
      this.attachParticipantTracks(participant, previewContainer);
    });

    // When a Participant joins the Room, log the event.
    room.on('participantConnected', participant => {
      console.log(participant);
      message.info('Joined');

      console.log("Joining: '" + participant.identity + "'");
    });

    // When a Participant adds a Track, attach it to the DOM.
    room.on('trackAdded', (track, participant) => {
      //this.log(participant.identity + " added track: " + track.kind);
      console.log(participant.identity + ' added track: ' + track.kind);

      var previewContainer = document.getElementById('remote-media');
      this.attachTracks([track], previewContainer);
    });

    // When a Participant removes a Track, detach it from the DOM.
    room.on('trackRemoved', (track, participant) => {
      console.log(participant.identity + ' removed track: ' + track.kind);
      this.detachTracks([track]);
    });

    // When a Participant leaves the Room, detach its Tracks.
    room.on('participantDisconnected', participant => {
      console.log("Participant '" + participant.identity + "' left the room");
      this.detachParticipantTracks(participant);
    });

    // Once the LocalParticipant leaves the room, detach the Tracks
    // of all Participants, including that of the LocalParticipant.
    room.on('disconnected', (room, error) => {
      if (error) {
        message.error(error);
      }
      console.log('Left');
      if (this.previewTracks) {
        this.previewTracks.forEach(track => {
          track.stop();
        });
      }
      this.detachParticipantTracks(room.localParticipant);
      room.participants.forEach(this.detachParticipantTracks.bind(this));
      console.log('room.sid---------------------------------------');
      console.log(room.sid);
      let _self = this;
      _self.setState({ sid: room.sid });

      });
  };
  attachTracks = (tracks, container) => {
    tracks.forEach(track => {
      container.appendChild(track.attach());
    });
  };

  attachParticipantTracks = (participant, container) => {
    var tracks = Array.from(participant.tracks.values());
    this.attachTracks(tracks, container);
  };

  detachTracks = tracks => {
    tracks.forEach(track => {
      track.detach().forEach(detachedElement => {
        detachedElement.remove();
      });
    });
  };

  detachParticipantTracks = participant => {
    var tracks = Array.from(participant.tracks.values());
    this.detachTracks(tracks);
  };

  log = message => {
    try {
      var logDiv = this.refs.log;
      logDiv.innerHTML += '<p>&gt;&nbsp;' + message + '</p>';
      logDiv.scrollTop = logDiv.scrollHeight;
    } catch (ex) {
      console.log(ex);
    }
  };
componentDidMount() {
    const _self = this;
    //console.log(this.props.currentQuestion.questionDescription);
    var connectOptions = {
      audio: true,
      name: 'Room' + Math.random(),
      video: { width: 640 },

      logLevel: 'debug',
    };
    setTimeout(() => {
      //      _self.setState({ loadingbutton: false });
      console.log('state changedd ');
    }, 15000);

    if (this.previewTracks) {
      connectOptions.tracks = this.previewTracks;
    }

    Video.connect(this.props.dataToken, connectOptions).then(this.roomJoined, error => {
      if (error) {
        let descriptionErr = '';
        if (error.message.includes('Only secure origins are allowed')) {
          descriptionErr = 'Only secure origins are allowed, Please use https instead http.';
        } else {
          descriptionErr = error.message;
        }

      } else {
        console.log('connected to twilio');
      }
      console.log('Could not connect to Twilio: ' + error.message);
    });
  }
  

`

when I see preview of recorded video then there first 3-4 sec video is missing.
I am using react js , ant design.

**Comment by manjeshbhargav** [comment]
Hi @maheshgiri ,

For issues related to recordings, please contact our [support](https://support.twilio.com/hc/en-us/requests/new?_ga=2.111421324.968573559.1587878324-1788440434.1576776383).
